var/spack/repos/builtin/packages/qhull/package.py

```diff
@@ -47,8 +48,7 @@ class Qhull(Package):
     
     depends_on('cmake')
 
-    def install(self, spec, prefix):
-        with working_dir('spack-build', create=True):
-            cmake('..', *std_cmake_args)
-            make()
-            make("install")
+    @CMakePackage.sanity_check('build')
+    @CMakePackage.on_package_attributes(run_tests=True)
```

Comment:
> Building in different directories, all of the ugly ALL_CAPS_FLAGS, and each enabled flag adds twenty subflags.

That's historical stuff for the most part. Different directories is a nice thing though, you really just need to be aware of whether you should use `SOURCE_DIR` or `BINARY_DIR`.